packages/typescript/mixin.core.js

```diff
@@ -28,29 +28,78 @@ const getCompilerOptions = (ts, rootPath) => {
 
 class TypescriptMixin extends Mixin {
   configureBuild(webpackConfig, { jsLoaderConfig, allLoaderConfigs }, target) {
-    const { loader, options, exclude } = jsLoaderConfig;
-    const isDevelop =
-      target === 'develop' ||
-      (target === 'node' && process.env.NODE_ENV !== 'production');
-    const loaderOptions = isDevelop
-      ? { compilerOptions: { isolatedModules: true }, transpileOnly: true }
-      : undefined;
+    const { experimentalEsbuild } = this.options;
 
-    allLoaderConfigs.unshift({
-      test: /\.tsx?$/,
-      exclude,
-      use: [
+    webpackConfig.resolve.extensions.push('.ts', '.tsx');
+
+    if (experimentalEsbuild) {
+      const {
+        include,
+        exclude,
+        use: [{ loader, options }, ...loaders],
```

Comment:
```suggestion
        use: [{ loader: esbuildLoader, options }, ...loaders],
```

To make it a little more clear what's going on here.